src/main/java/com/omar/employee/controller/EmployeeController.java

```diff
@@ -45,8 +46,13 @@ public ResponseEntity<String> addEmployee(@RequestBody EmployeeRequest request)
             return new ResponseEntity<>(HttpStatus.OK);
         }
 
-        String employeeId = ((SrvHandlerAddEmployee) srvHandlerMap.get(SrvHandlerVCB.ADD_EMPLOYEE)).addEmployee(request);
-        return new ResponseEntity<>(employeeId, HttpStatus.CREATED);
+        try {
+            String employeeId = ((SrvHandlerAddEmployee) srvHandlerMap.get(SrvHandlerVCB.ADD_EMPLOYEE)).addEmployee(request);
+            return new ResponseEntity<>(employeeId, HttpStatus.CREATED);
+        }catch (EmployeeDataNotFoundException e) {
+            String errorMessage = "Error: " + e.getMessage();

```

Comment:
can we print informative log about what is the missing data ? example bad mail ?